evcouplings/complex/protocol.py

```diff
@@ -8,15 +8,159 @@
   Anna G. Green
 """
 
+import numpy as np
+import pandas as pd
+from collections import Counter
+
 from evcouplings.couplings.mapping import Segment
+
 from evcouplings.utils.config import (
-    check_required, InvalidParameterError,
-    read_config_file, write_config_file
+    check_required, InvalidParameterError
 )
 from evcouplings.utils.system import (
-    create_prefix_folders, valid_file,
-    verify_resources,
+    create_prefix_folders, verify_resources
+)
+from evcouplings.align.protocol import modify_alignment
+from evcouplings.align.alignment import Alignment
+from evcouplings.align.ids import retrieve_sequence_ids
+
+from evcouplings.complex.alignment import (
+    write_concatenated_alignment
 )
+from evcouplings.complex.distance import (
+    find_possible_partners, best_reciprocal_matching,
+    plot_distance_distribution
+)
+from evcouplings.complex.similarity import (
+    read_species_annotation_table,
+    most_similar_by_organism,
+    filter_best_reciprocal,
+    find_paralogs
+)
+
+
+def describe_concatenation(annotation_file_1, annotation_file_2,
+                           genome_location_filename_1, genome_location_filename_2,
+                           outfile):
+    """
+    Describes properties of concatenated alignment. 
+
+    Writes a csv with the following columns
+
+    num_seqs_1 : number of sequences in the first monomer alignment
+    num_seqs_2 : number of sequences in the second monomer alignment
+    num_nonred_species_1 : number of unique species annotations in the 
+        first monomer alignment
+    num_nonred_species_2 : number of unique species annotations in the 
+        second monomer alignment
+    num_species_overlap: number of unique species found in both alignments
+    median_num_per_species_1 : median number of paralogs per species in the 
+        first monomer alignmment
+    median_num_per_species_2 : median number of paralogs per species in 
+        the second monomer alignment
+    num_with_embl_cds_1 : number of IDs for which we found an EMBL CDS in the 
+        first monomer alignment (relevant to distance concatention only)
+    num_with_embl_cds_2 : number of IDs for which we found an EMBL CDS in the 
+        first monomer alignment (relevant to distance concatention only)
+    
+    Parameters
+    ----------
+    annotation_file_1 : str
+        Path to annotation.csv file for first monomer alignment
+    annotation_file_2 : str
+        Path to annotation.csv file for second monomer alignment
+    genome_location_filename_1 : str
+        Path to genome location mapping file for first alignment
+    genome_location_filename_2 : str
+        PPath to genome location mapping file for second alignment
+    outfile: str
+        Path to output file
+    """
+
+    # load the annotations for each alignment
+    # as {id:species}
+    annotations_1 = read_species_annotation_table(
+        annotation_file_1
+    )
+    species_1 = list(annotations_1.species)
```

Comment:
btw if you want a list representation you could also use annotations_1.species.values which gives back the underlying numpy array